src/Core/NETPortable/NetPortableProfileTable.cs

```diff
@@ -14,95 +13,95 @@ public static class NetPortableProfileTable
     {
         private const string PortableReferenceAssemblyPathEnvironmentVariableName = "NuGetPortableReferenceAssemblyPath";
 
-        // This collection is the original indexed collection where profiles are indexed by 
-        // the full "ProfileXXX" naming. 
-        private static NetPortableProfileCollection _portableProfiles;
-        // In order to make the NetPortableProfile.Parse capable of also parsing so-called 
-        // "custom profile string" version (i.e. "net40-client"), we need an alternate index
-        // by this key. I used dictionary here since I saw no value in creating a custom collection 
-        // like it's done already for the _portableProfiles. Not sure why it's done that way there.
-        private static IDictionary<string, NetPortableProfile> _portableProfilesByCustomProfileString;
-        // key is the identifier of the optional framework and value is the list of tuple of (optional Framework Version, set of profiles in which they are optional)
-        private static IDictionary<string, List<VersionStringISetTuple>> _portableProfilesSetByOptionalFrameworks;
+        private static Lazy<CompiledNetPortableProfileCollection> _lazyPortableProfiles
+            = new Lazy<CompiledNetPortableProfileCollection>(() => new CompiledNetPortableProfileCollection(BuildPortableProfileCollection()));
+        private static CompiledNetPortableProfileCollection _portableProfiles;
+
+        private static CompiledNetPortableProfileCollection Compiled
+        {
+            get
+            {
+                return _portableProfiles ?? _lazyPortableProfiles.Value;
+            }
+        }
 
         public static NetPortableProfile GetProfile(string profileName)
         {
-            if (String.IsNullOrEmpty(profileName))
+            if (string.IsNullOrEmpty(profileName))
             {
                 throw new ArgumentException(CommonResources.Argument_Cannot_Be_Null_Or_Empty, "profileName");
             }
 
             // Original behavior fully preserved, as we first try the original behavior.
             // NOTE: this could be a single TryGetValue if this collection was kept as a dictionary...
-            if (Profiles.Contains(profileName))
+            if (Compiled.Profiles.Contains(profileName))
```

Comment:
That's not thread-safe at all (inconsistent). Each time in 3 member access occurrences here you may get different instances. Consider moving the method to the private class or cache the instance at the beginning to a local variable.